cvprac/cvp_api.py

```diff
@@ -3947,12 +3949,16 @@ def svc_account_token_get_one(self, token_id):
                       'time': '2022-05-03T15:38:53.725014447Z', 'type': 'INITIAL'}]
         '''
         msg = 'Service Account Resource APIs are supported from 2021.3.0+.'
-        if self.cvp_version_compare('>=', 7.0, msg):
-            payload = {"key": {"id": token_id}}
-            url = '/api/v3/services/arista.serviceaccount.v1.TokenService/GetOne'
-            self.log.debug(f"v7 {url} {payload}")
-            return self.clnt.post(url, data=payload)
-        return None
+        endpoint_legacy = '/api/v3/services/arista.serviceaccount.v1.TokenService/GetOne'
+        endpoint = '/api/resources/serviceaccount/v1/Token'
+        payload = {"key": {"id": token_id}}
+        if self.cvp_version_compare('>=', 12.0, msg):
+            query_param = f"?key.id={token_id}"
+            self.log.debug(f'v12 {endpoint + query_param}')
+            return self.clnt.post(endpoint + query_param)
+        elif self.cvp_version_compare('>=', 7.0, msg) and self.cvp_version_compare('<', 12.0, msg):
```

Comment:
We can remove the check for less than V12 since greater or equal is handled in the previous condition.